lib/rucio/client/uploadclient.py

```diff
@@ -303,9 +303,13 @@ def _pick_random_rse(rse_expression):
                 if not no_register:
                     if register_after_upload:
                         self._register_file(file, registered_dataset_dids)
-                    replica_for_api = self._convert_file_for_api(file)
-                    if not self.client.update_replicas_states(rse, files=[replica_for_api]):
-                        logger(logging.WARNING, 'Failed to update replica state')
+                    else:
+                        replica_for_api = self._convert_file_for_api(file)
+                        try:
+                            self.client.update_replicas_states(rse, files=[replica_for_api])
+                        except Exception as error:
```

Comment:
I'm not sure if this should be considered a fatal error or not. Some other places in this routine only emit warnings for what seem to me fatal errors.

Reply:
Hmmmm, I concur, fatal seems the appropriate one. @bari12 opinion?

Reply:
Yes, should be fatal.

Reply:
@nsmith- ping 🔔

Reply:
ah sorry for the delay, I changed warning -> error. Let me know if something more needs to be done, its not clear to me why not all cases are errors, e.g. line 288 or 319 have only warning